lib/migration.pm

```diff
@@ -157,4 +158,26 @@ sub reset_consoles_tty {
     reset_consoles;
 }
 
+# Assert read-only snapshot before migrated
+# Assert screen 'linux-login' with 200s
+# Workaround known issue: bsc#980337
+# In this case try to select tty1 with multi-times then select root console
+sub boot_into_ro_snapshot {
+    if (!check_screen('linux-login', 200)) {
+        record_soft_failure 'bsc#980337';
+        my $retries = 10;
+        for my $i (0 .. $retries) {
+            send_key "ctrl-alt-f1";
+            if (check_screen('tty1-selected', 5)) {
+                last;
+            }
+            else {
+                record_info('not in tty1', 'switch to tty1 failed', result => 'softfail');
+            }
+            die "Boot into read-only snapshot failed over 5 minutes, considering a product issue" if $i == $retries;
```

Comment:
You could simply put this out of the loop unconditionally to avoid check on every iteration and return once tty1-selected matches.

Reply:
yeah, i have optimized a little